### ISSUE TITLE
How is the y-axis (Shap Interaction value) in the generated plot calculated?

### ISSUE THREAD

**Issue by SharkFin-top** [MAIF/shapash#477]
Thank you for the work of shapash, it is a great Python package!

I have a question when using `xpl.plot.interactions_plot`: **How is the y-axis (Shap Interaction value) in the generated plot calculated?**

I noticed that it is not the value of the feature represented by the x-axis or its Shap value.
It is also not the value of the feature represented by the color or its Shap value.

I would greatly appreciate your response. Thanks♪(･ω･)ﾉ

**Comment by ThomasBouche** [comment]
Hi, Thanks,

The shap interaction value is compute by shap. 
if you want to know more, I recommend this notebook and this book:
- https://github.com/shap/shap/blob/master/notebooks/tabular_examples/tree_based_models/NHANES%20I%20Survival%20Model.ipynb
- https://christophm.github.io/interpretable-ml-book/shap.html#shap

**Comment by ThomasBouche** [comment]
if you have compile xpl.plot.top_interactions_plot(), you have interactions values in this attributes of the SmartExplainer:
```
xpl.interaction_values
```

**Comment by SharkFin-top** [comment]
> if you have compile xpl.plot.top_interactions_plot(), you have interactions values in this attributes of the SmartExplainer:
> 
> ```
> xpl.interaction_values
> ```

Thank U very much!!!

**Comment by SharkFin-top** [comment]
@ThomasBouche 

Thank you for your response, I have encountered a problem that has confused me: the y-axis values of the plot generated by `xpl.interactions_plot` are always **half** of the y-axis values of `shap.dependence_plot`

For example, I am using the classic Boston dataset as an example.

Below is a snippet of my code.

First, I imported the necessary packages and data, and performed some basic machine learning.

```python
# package
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from xgboost import XGBRegressor
from sklearn.model_selection import train_test_split
import shap

# ----------
# dataset
from sklearn.datasets import load_boston
boston = load_boston()
df = pd.DataFrame(boston.data, columns=boston.feature_names)
df['MEDV'] = boston.target

# ----------
# ML
X = df.drop('MEDV', axis=1)
y = df['MEDV']
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=123)
model = XGBRegressor()
model.fit(X_train, y_train)
y_pred = model.predict(X_test)

```



Next, I used `shap` in Python to analyze the interactions.

```python
shap_interaction_values = shap.TreeExplainer(model).shap_interaction_values(X_test)
```



I use `shap.dependence_plot` to analyze the interaction between two variables.

```python
shap.dependence_plot(('INDUS', 'ZN'), shap_interaction_values, X_test)
```

![image](https://github.com/MAIF/shapash/assets/55095619/ccd97c1a-0ff4-4e41-a4b7-efc14aceb220)



```python
shap.dependence_plot(('AGE', 'TAX'), shap_interaction_values, X_test)
```

![image](https://github.com/MAIF/shapash/assets/55095619/ee49b320-7037-4357-a221-f03013f0807e)





Then, I use `shapash` in Python for interaction analysis.

```python
from shapash import SmartExplainer

xpl = SmartExplainer(model=model)

xpl.compile(x=X_test)
```



I use `xpl.interactions_plot` to analyze the interaction of 2 variables.

```
xpl.plot.interactions_plot('INDUS', 'ZN')
```

![image](https://github.com/MAIF/shapash/assets/55095619/45e11908-4b8c-47ce-8212-592bcb0fe5b8)



```python
xpl.plot.interactions_plot('AGE', 'TAX')
```

![image](https://github.com/MAIF/shapash/assets/55095619/d158d624-0fd9-4fc9-abbb-e6cc5519a72c)



---

Now, let's compare the results.

![image](https://github.com/MAIF/shapash/assets/55095619/0daa5dca-2fb2-4ff8-94e0-5b2026cb252a)

I was surprised to find that the distribution of the scatter points and the x-axis are the same when comparing the plots generated by `xpl.interactions_plot` and `shap.dependence_plot`.

However, the values on the y-axis are different. the y-axis values of the plot generated by `xpl.interactions_plot` are always **half** of the y-axis values of `shap.dependence_plot`



I really don't understand why this is happening. Maybe there's an issue with my code? I would greatly appreciate a response, thank you!

**Comment by SharkFin-top** [comment]
thank you for your working!



---Original---
From: ***@***.***&gt;
Date: Thu, Aug 24, 2023 16:57 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [MAIF/shapash] How is the y-axis (Shap Interaction value) in thegenerated plot calculated? (Issue #477)




 
Hi,
 It's a good question, and the problem is with Shapash.
 Shapash doesn't multiply the matrix interactions by 2 to account for interactions between 2 variables.
 

 
In this example, the value displayed is -0.245, when it should be double.
 
I will create an issue to correct this
